src/Resources/Resources.Test/ScenarioTests/DeploymentTests.ps1

```diff
@@ -146,6 +146,111 @@ function Test-NewDeploymentFromTemplateSpec
     }
 }
 
+function Test-NewSubscriptionDeploymentFromTemplateSpec
```

Comment:
I'd also add a negative test case where we expect a deployment to fail with given TS.

Reply:
Ok. I will add them.

Reply:
Ideally, the negative test case would look for non existing TS or TS version. However currently this scenario doesn't produce a meaningful error message. There is a task in the board to fix that, once it's fixed, Gokul will update the test case.